jujugui/static/gui/src/test/test_app.js

```diff
@@ -909,18 +911,37 @@ describe('App', function() {
         // Disconnect and reconnect the WebSocket.
         conn.transient_close();
         conn.open();
-        assert.equal(1, conn.messages.length);
+        assert.equal(1, conn.messages.length, 'no login messages sent');
         assertIsLogin(conn.last_message());
         done();
       });
     });
 
+    it('tries to re-login with macaroons on disconnections', function(done) {
+      sessionStorage.clear();
```

Comment:
The area that is storing this stuff in localStorage should really have the ability to pass in a stub of localStorage as I requested here as well https://github.com/juju/juju-gui/pull/1602#discussion_r61636556 This can be in a follow-up I guess but it's a pretty leaky thing to be using the real global storage in tests.